include/mqt-core/dd/Edge.hpp

```diff
@@ -51,7 +51,8 @@ using MatrixEntryFunc = std::function<void(const std::size_t, const std::size_t,
  */
 template <class Node> struct Edge {
   Node* p;
-  Complex w;
+  Complex w{};
+  std::size_t numQubits{};
```

Comment:
Just a quick side comment here:
This change is not really acceptable as it increases the size of _every_ single node in the decision diagrams (the literal nodes, not the `Node` class) by 33%. For something that is not even needed for the main functionality (MxV multiplication, etc.), this is a tradeoff that we cannot make here.

An acceptable change would be to adjust the signature of the functions that explicitly need the number of qubits to take an additional argument. This is closely related to the other comment in this PR about the potential need for a new level of abstraction above the `Edge` class that, e.g., has the information on the number of qubits. Although I would assume that this new level of abstraction is more work than simply adjusting the function signatures and it might even make some things a little more challenging (e.g., dropping the number of qubits member from the `Operation` class).
Bottomline: I would just revise the signatures of the couple of functions that need the number of qubits.

Reply:
I just fixed it right now. Thanks a lot for the advice!